src/main/java/de/dhbw/karlsruhe/Main.java

```diff
@@ -1,22 +1,29 @@
 package de.dhbw.karlsruhe;
 
+import de.dhbw.karlsruhe.services.LogoutService;
 import de.dhbw.karlsruhe.services.MenuDialogService;
 import de.dhbw.karlsruhe.services.StartUpDialogService;
 
+import java.util.Scanner;
+
 public class Main {
 
 	public static void main(String[] args) {
+		
+		LogoutService logoutService = new LogoutService();
 		StartUpDialogService startUpDialogService = new StartUpDialogService();
         MenuDialogService menuDialogService = new MenuDialogService();
 
-		boolean successfulLogin = startUpDialogService.signIn();
+		while (!logoutService.getSignedIn()) {
+			boolean successfulLogin = startUpDialogService.signIn();
 
-		if (successfulLogin) {
-			System.out.println("You have successfully logged in!");
-            menuDialogService.startMenuDialog();
-		} else {
-			System.out.println("Login failed!");
+			if (successfulLogin) {
+				logoutService.setSignedIn(true);
```

Comment:
Anstatt successfulLogin den Rückgabewert von startupDialogServiec.signIn() zuzuweisen, kann der Wert direkt in logoutService.setSignedIn gesetzt werden. Dann kann das if-Statement gespart werden, wenn der MenueDialogService.startMenuDialog() Aufruf erst nach der while Schleife stattfindet.  Die while-Schleife ist auch gar nicht nötig, da im startUpDialogService schon solange mit while gelooped wird, bis der Login erfolgreich war.
Außerdem brauchen wird den Dialog vom User hier nicht, da der startUpDialogService bereits Text bei Erfolg und nicht Erfolg ausgibt.
Mein Vorschlag wäre also:
```
logoutService.setSignedIn(startUpDialogService.signIn());
 menuDialogService.startMenuDialog();
```
Anstelle von:
```
while (!logoutService.getSignedIn()) {
			boolean successfulLogin = startUpDialogService.signIn();

			if (successfulLogin) {
				logoutService.setSignedIn(true);
				System.out.println("You have successfully logged in!");
				menuDialogService.startMenuDialog();
			} else {
				System.out.println("Login failed!");
			}
		}
```

Reply:
Guter Hinweis, danke dir!

Durch die while Schleife ist es möglich, sich auch nach dem ausloggen erneut einzuloggen. Das könnte hilfreich sein, wenn man z.B. nur den Benutzer wechseln will.
Über die Funktionalität haben wir uns noch nicht genauer ausgetauscht.

Mein Vorschlag ist, dass wir den Benutzer einfach nochmal fragen, ob er sich nochmal einloggen oder das Programm beenden will.

Reply:
Falls wir uns auf ein anderes Verhalten einigen, ist der Code mit der neuen Änderung schneller anpassbar